tests/ballerina-spec-conformance-tests/src/test/java/org/ballerinalang/test/TestRunnerUtils.java

```diff
@@ -122,6 +122,7 @@ private static void readTestFile(File testFile, String tempDir, String fileName,
             testCase[6] = isSkippedTestCase;
             testCase[8] = headersOfTestCase.containsKey(FAIL_ISSUE);
             testCase[9] = headersOfTestCase.get(LABELS);
+            testCase[10] = Arrays.asList("transactional-expr");
```

Comment:
Why do we send `skippedTestLabels` through the data provider(any special reason for this)? Can't we define `skippedTestLabels` within `validateLabels` function?

Reply:
not really. thought that was cleaner since necessary details are passed through the data provider.

Reply:
not really. thought that was cleaner since necessary details are passed through the data provider.

Reply:
Hmmm. My initial thought was instead of updating the test runner manually by the user; the use of a separate file as skip label input will be good. As we use the data provider to get the inputs by parsing the test files; we can use the same place to parse the skip label input file.  

But maybe let's not add the skip label list at the provider. Instead for the moment, let's maintain a module-level var to hold the set. We can decide how we populate that later(whether it's read from a file or updated by the user manually) and use that set at the `isSkippedTestCase()` not at `validateLabels` to mark it as skipped for the skip labeled.

Reply:
+1. will update.

Reply:
What I feel is, data provider gives a summary of the user input, things like predefinedLabels, skippedLabels are not part of the user input; rather it is a part of the test runner test execution. Let's wait for @pcnfernando's thoughts as well.

Reply:
sorry I missed the previous comment. :)